recipes/lora_finetune_distributed.py

```diff
@@ -577,27 +578,31 @@ def save_checkpoint(
         - Merged weights with key MODEL_KEY
         - Adapter weights with key ADAPTER_KEY
         - Relevant recipe state if training is not complete
-        - If the `self._save_adapter_weights_only` option is True, the checkpointer will save only the adapter weights
 
-        To correctly resume from training, the adapter weights and recipe state must be provided along with the base model weights.
+        Checkpointer will save the merged weights, adapter weights and recipe state in
+        different checkpoint files. To correctly resume from training, the adapter weights
+        and recipe state must be provided along with the base model weights.
         """
         # final dict passed onto the checkpointer
         checkpoint_dict = {}
 
         intermediate_checkpoint = epoch + 1 < self.total_epochs
         # To prevent GPU memory from spiking during checkpoint save,
         # we consolidate the full model and optim state dicts on CPU for rank 0
-        with FSDP.state_dict_type(
+        cpu_state_dict = training.get_full_model_state_dict(
             self._model,
-            StateDictType.FULL_STATE_DICT,
-            FullStateDictConfig(offload_to_cpu=True, rank0_only=True),
-            FullOptimStateDictConfig(offload_to_cpu=True, rank0_only=True),
-        ):
-            cpu_state_dict = self._model.state_dict()
-            if intermediate_checkpoint:
-                opt_state_dict = FSDP.optim_state_dict(self._model, self._optimizer)
-            else:
-                opt_state_dict = None
+            self._is_rank_zero,
+            device=self._device,
+        )
+
+        if intermediate_checkpoint:
```

Comment:
Did you port over the ability to save adapter weights only without consolidating the full model checkpoint first from the state dict?

Reply:
Do we do that today? Iiuc it seems like we don't? [ref](https://github.com/pytorch/torchtune/blob/66590b408b64fcff32a8b75b84f592b4e1530a00/recipes/lora_finetune_distributed.py#L590-L652)

Reply:
Sorry, I forgot that's only in the 405B PR